pages/recording/[[...id]].tsx

```diff
@@ -158,25 +158,22 @@ function RecordingPage({
         return;
       }
 
-      setRecording(rec);
-
-      const isTestReplay = rec.metadata?.test && rec.metadata?.source;
-      const isTestWorkspace = rec.workspace?.isTest;
-
-      if (rec.private) {
-        if (isTestReplay && !isTestWorkspace) {
-          setExpectedError({
-            content: "This recording is not available.",
-            message: "The recording must belong to a test suite",
-          });
-        } else if (!isTestReplay && isTestWorkspace) {
-          setExpectedError({
-            content: "This recording is not available.",
-            message: "The recording cannot be in a test suite",
-          });
-        }
+      if (rec.isTest && !rec.isInTestWorkspace) {
+        setExpectedError({
+          content: "This recording is not available.",
+          message: "The recording must belong to a test suite",
+        });
+        return;
+      } else if (!rec.isTest && rec.isInTestWorkspace) {
+        setExpectedError({
+          content: "This recording is not available.",
+          message: "The recording cannot be in a test suite",
+        });
```

Comment:
I don't think the frontend should be doing this check anymore either.

After BAC-3531, the backend should reject any attempts to open a test recording within a non-test workspace (and vice versa).

Reply:
The backend will refuse to create a session for test recordings in a non-test workspace and vice versa. But I think we should still keep the check here to show the user a more specific error message explaining why the recording can't be viewed.

Reply:
I guess, but I would expect the backend to inform the frontend of what that message should be (with a new error code if needed). It seems like we might guess incorrectly, otherwise.